validator/stateless_block_validator.go

```diff
@@ -307,11 +307,17 @@ func SetMachinePreimageResolver(ctx context.Context, mach *ArbitratorMachine, pr
 			if err != nil {
 				log.Error("Failed to deserialize DAS message", "err", err)
 			} else {
-				dasPreimage, err := das.Retrieve(ctx, seqMsg[40:])
+				keysetPreimage, err := das.KeysetFromHash(ctx, cert.KeysetHash[:])
 				if err != nil {
-					return fmt.Errorf("couldn't retrieve message from DAS %w", err)
+					log.Error("Couldn't get keyset from DAS", "err", err)
```

Comment:
I think in this case verification will definitely fail and we should error here rather than fail with a missing preimage during verification. The L1 should guarantee that we can get the preimage for any valid keyset hash